cdap-distributions/src/packer/cdap-sdk-ubuntu12-with-uri.json

```diff
@@ -0,0 +1,165 @@
+{
+  "variables": {
+    "sdk_version": "3.3.0"
+  },
+  "builders": [
+    {
+      "type": "virtualbox-iso",
+      "guest_os_type": "Ubuntu_64",
+      "iso_url": "http://releases.ubuntu.com/12.04/ubuntu-12.04.5-server-amd64.iso",
+      "iso_checksum": "769474248a3897f4865817446f9a4a53",
+      "iso_checksum_type": "md5",
+      "ssh_username": "root",
+      "ssh_password": "cdap",
+      "ssh_wait_timeout": "30m",
+      "shutdown_command": "shutdown -P now",
+      "format": "ova",
+      "headless": true,
+      "guest_additions_mode": "upload",
+      "http_directory": "files",
+      "boot_command": [
+        "<esc><esc><enter><wait>",
+        "/install/vmlinuz noapic ",
+        "preseed/url=http://{{ .HTTPIP }}:{{ .HTTPPort }}/preseed.cfg ",
+        "debian-installer=en_US auto locale=en_US kbd-chooser/method=us ",
+        "hostname={{ .Name }}.cask.co ",
+        "fb=false debconf/frontend=noninteractive ",
+        "keyboard-configuration/modelcode=SKIP keyboard-configuration/layout=USA ",
+        "keyboard-configuration/variant=USA console-setup/ask_detect=false ",
+        "initrd=/install/initrd.gz -- <enter>"
+      ],
+      "vboxmanage": [
+        ["modifyvm", "{{ .Name }}", "--memory", "4096"]
+      ],
+      "vboxmanage_post": [
+        ["modifyvm", "{{ .Name }}", "--clipboard", "bidirectional"],
+        ["modifyvm", "{{ .Name }}", "--accelerate2dvideo", "on"],
+        ["modifyvm", "{{ .Name }}", "--accelerate3d", "on"],
+        ["modifyvm", "{{ .Name }}", "--natpf1", "guest_ssh,tcp,127.0.0.1,2222,,22" ]
+      ],
+      "vm_name": "cdap-standalone-vm-{{user `sdk_version`}}",
+      "name": "cdap-sdk-vm"
+    }
+  ],
+  "provisioners": [
+    {
+      "type": "shell",
+      "scripts": [
+        "scripts/apt-setup.sh",
+        "scripts/cookbook-dir.sh"
+      ],
+      "pause_before": "10s"
+    },
+    {
+      "type": "shell",
+      "inline": [
+        "reboot"
+      ],
+      "only": ["cdap-sdk-vm"]
```

Comment:
The intent was to use this to eventually also create Docker containers. However, the version of Packer that we've been using doesn't support ENTRYPOINT, which is how we start CDAP in our current Docker container. If I changed how that worked, and forced the SDK service to start somehow, I planned on pulling Docker into this file. The VM-specific stuff has been marked as such, where the "common" things between Docker/VM are not marked.